apps/dashr-web-trader/app.R

```diff
@@ -0,0 +1,1612 @@
+    state("USDCHFchart", "figure"),
+    state("USDCHFgraph_div", "style")
+  ),
+  function(n_i, p, t, s, pairs, a, b, old_fig, display){
+    pairs <- unlist(strsplit(pairs, ","))
+    if ("USDCHF" %in% pairs){
+      f <- getFig(USDCHF, a, b, t, s, p)
+      data <- f$x$data
+      layout <- f$x$layout
+      layout["uirevision"] <- display$position
+      layout[["xaxis"]]["tickformat"] <- "%H:%M"
+      layout[["margin"]] <- list(b = "20%", r = "0", l = "50", t = "0")
+      layout[["xaxis"]]["showgrid"] <- FALSE
+      layout["paper_bgcolor"] <- "#262a30"
+      layout["plot_bgcolor"] <- "#262a30"
+      layout["autosize"] <- TRUE
+      return(list(data = data, layout = layout))
+    }
+    return(list(data = list(), layout = list()))
+  }
+)
+app$callback(
+  output("USDJPYchart", "figure"),
+  list(
+    input("i_tris", "n_intervals"),
+    input("USDJPYdropdown_period", "value"),
+    input("USDJPYchart_type", "value"),
+    input("USDJPYstudies", "values"),
+    input("charts_clicked", "children"),
+    state("USDJPYask", "children"),
+    state("USDJPYbid", "children"),
+    state("USDJPYchart", "figure"),
+    state("USDJPYgraph_div", "style")
+  ),
+  function(n_i, p, t, s, pairs, a, b, old_fig, display){
+    pairs <- unlist(strsplit(pairs, ","))
+    if ("USDJPY" %in% pairs){
+      f <- getFig(USDJPY, a, b, t, s, p)
+      data <- f$x$data
+      layout <- f$x$layout
+      layout["uirevision"] <- display$position
+      layout[["xaxis"]]["tickformat"] <- "%H:%M"
+      layout[["margin"]] <- list(b = "20%", r = "0", l = "50", t = "0")
+      layout[["xaxis"]]["showgrid"] <- FALSE
+      layout["paper_bgcolor"] <- "#262a30"
+      layout["plot_bgcolor"] <- "#262a30"
+      layout["autosize"] <- TRUE
+      return(list(data = data, layout = layout))
+    }
+    return(list(data = list(), layout = list()))
+  }
+)
+app$callback(
+  output("GBPUSDchart", "figure"),
+  list(
+    input("i_tris", "n_intervals"),
+    input("GBPUSDdropdown_period", "value"),
+    input("GBPUSDchart_type", "value"),
+    input("GBPUSDstudies", "values"),
+    input("charts_clicked", "children"),
+    state("GBPUSDask", "children"),
+    state("GBPUSDbid", "children"),
+    state("GBPUSDchart", "figure"),
+    state("GBPUSDgraph_div", "style")
+  ),
+  function(n_i, p, t, s, pairs, a, b, old_fig, display){
+    pairs <- unlist(strsplit(pairs, ","))
+    if ("GBPUSD" %in% pairs){
+      f <- getFig(GBPUSD, a, b, t, s, p)
+      data <- f$x$data
+      layout <- f$x$layout
+      layout["uirevision"] <- display$position
+      layout[["xaxis"]]["tickformat"] <- "%H:%M"
+      layout[["margin"]] <- list(b = "20%", r = "0", l = "50", t = "0")
+      layout[["xaxis"]]["showgrid"] <- FALSE
+      layout["paper_bgcolor"] <- "#262a30"
+      layout["plot_bgcolor"] <- "#262a30"
+      layout["autosize"] <- TRUE
+      return(list(data = data, layout = layout))
+    }
+    return(list(data = list(), layout = list()))
+  }
+)
+
+# Generate_open_close_menu_callback
```

Comment:
https://github.com/plotly/dash-sample-apps/blob/86cbe74e7e152e7db6da030de7f844d8e08a5f52/apps/dashr-web-trader/app.R#L552-L612

Above 4 callbacks can be removed. Since below callbacks are implementing the same functionality

https://github.com/plotly/dash-sample-apps/blob/86cbe74e7e152e7db6da030de7f844d8e08a5f52/apps/dashr-web-trader/app.R#L939-L1007

Reply:
Agreed, I'll remove them